cirq/qaoa/qaoa_function.py

```diff
@@ -0,0 +1,118 @@
+# Cirq Quantum Approximate Optimization
+#
+# Fabio Sanches
+
+# imports
+import typing
+import numpy as np
+import QAOA
+
+from typing import Union
+
+def qaoa_solver(n_qubits,
+                objective_function,
+                optimization_method,
+                p,
+                mixing = 'X', expectation_method = 'wavefunction'):
+    """
+    QAOA solver helper function
+
+    :param n_qubits: number of qubits
+    :param objective_function: objective function to be optimized
+        type should be Dict[PauliString: float]
+    :param optimization_method: type of optimizer used, currently available are
+        swarm, nelder-mead, bayesian
+    :param p: number of gamma and beta parameters/number of repeated
+        applications of unitaries from QAOA paper
+    :param mixing: mixing operator 'B' in QAOA paper. If standard 'X' on every
+        qubit, set to 'X', otherwise specify with type Dict[PauliString: float]
+
+    :return: dict with solution and gamma and beta values for optimal solution
+    """
+
+    qaoa = QAOA.QAOA(n_qubits=n_qubits,
+                objective_function=objective_function,
+                mixing_operator=mixing)
+
+    if optimization_method == 'swarm':
+
```

Comment:
This three conditional blocks are extremely similar. It looks like a dictionary mapping input method to optimizer method would be sufficient to combine them.